code/ui/components/src/tabs/tabs.helpers.tsx

```diff
@@ -10,25 +12,41 @@ export const VisuallyHidden = styled.div<VisuallyHiddenProps>(({ active }) =>
   active ? { display: 'block' } : { display: 'none' }
 );
 
-export const childrenToList = (children: any, selected: string) =>
+export const childrenToList = (children: TabsProps['children']) =>
   Children.toArray(children).map(
-    ({ props: { title, id, color, children: childrenOfChild } }: ReactElement, index) => {
-      const content = Array.isArray(childrenOfChild) ? childrenOfChild[0] : childrenOfChild;
+    ({
+      props: { title, id, color, children: childrenOfChild },
+    }: ReactElement<{
+      children: FC<Addon_RenderOptions> | ReactChild | null;
+      title: ReactChild | null | FC;
```

Comment:
Should this type match the one defined in `Addon_BaseType`? https://github.com/storybookjs/storybook/pull/23288/files#diff-66fd0ef2f3df056f95276eb1d7f4ba662f32a1ed1a494f0b6c932ef1da9a832aR317

Reply:
I don't think so. Tabs component is also used for non-panels.

Like I mention somewhere in a code-comment the Tabs component is a mess, and needs a complete overhaul, this makes an attempt at making it better within the constraints of not changing the API.